lib/tree.rb

```diff
@@ -1,38 +1,61 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
-
+class Tree
+  attr_accessor :height, :age, :apples, :alive
+  
   def initialize
+    @height = 0
+    @age = 0
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age += 1
+    @height += 1
+    add_apples
+    if @age >= 100 then @alive = false end
   end
 
   def add_apples
+    apples_to_grow = @age * 5
+    
+    apples_to_grow.times do
+      @apples << grown_apple
+    end
+  end
+
+  def grown_apple
+    Apple.new("red", rand(5..15))
   end
 
   def any_apples?
+    !@apples.empty?
   end
 
   def pick_an_apple!
     raise NoApplesError, "This tree has no apples" unless self.any_apples?
+    @apples.pop
   end
 
   def dead?
+    !@alive
   end
 end
 
 class Fruit
+  attr_accessor :has_seeds
   def initialize
-    has_seeds = true
+    @has_seeds = true
   end
 end
 
-class Apple <
-  attr_reader #what should go here 
+class Apple < Fruit
+  attr_accessor :color, :diameter
 
   def initialize(color, diameter)
+    super()
```

Comment:
This line caused me a few minutes of scratching my head, as "super" tried to call Fruit by passing through the two parameters, which raised an error since Fruit only has a nullary initialize method. I was assuming that "super" would just call the nullary constructor and not try to pass in any parameters as other methods in ruby don't require parenthesis when called!